coredb-operator/src/extensions.rs

```diff
@@ -0,0 +1,27 @@
+use crate::{Context, CoreDB, Error};
+use std::sync::Arc;
+use tracing::debug;
+
+pub async fn create_extensions(cdb: &CoreDB, ctx: &Arc<Context>) -> Result<(), Error> {
+    let client = &ctx.client;
+    let extensions = &cdb.spec.enabledExtensions;
+
+    // TODO(ianstanton) Some extensions will fail to create. We need to handle and surface any errors.
+    //  Logging result at debug level for now.
+
+    // iterate through list of extensions and run CREATE EXTENSION <extension-name> for each
+    for ext in extensions {
+        debug!("Creating extension: {}", ext);
+        // this will no-op if we've already created the extension
+        let result = cdb
+            .psql(
+                format!("CREATE EXTENSION {};", ext),
+                "postgres".to_owned(),
+                client.clone(),
+            )
+            .await
+            .unwrap();
```

Comment:
Not immediately a concern, but we may want to evaluate this for sql injection vulnerability in the future. i.e. what happens if an extension named `; drop from pg_catalog` gets through?

Reply:
I think in this case, we could validate it's a lower case alpha numerics and underscores

Reply:
I think what @sjmiller609 suggested works.

Reply:
Maybe longer term we bake something into `.psql()` to perform validation.